lint/unusedParam_checker.go

```diff
@@ -0,0 +1,60 @@
+package lint
+
+import (
+	"go/ast"
+)
+
+func init() {
+	addChecker(unusedParamChecker{}, &ruleInfo{})
+}
+
+type unusedParamChecker struct {
+	baseFuncDeclChecker
+}
+
+func (c unusedParamChecker) New(ctx *context) func(*ast.File) {
+	return wrapFuncDeclChecker(&unusedParamChecker{
+		baseFuncDeclChecker: baseFuncDeclChecker{ctx: ctx},
+	})
+}
+
+func (c *unusedParamChecker) CheckFuncDecl(decl *ast.FuncDecl) {
+	params := decl.Type.Params
+	if params == nil || params.NumFields() == 0 {
+		return
+	}
+
+	// collect all params to map
+	objToIdent := make(map[*ast.Object]*ast.Ident)
+	for _, p := range params.List {
+		if len(p.Names) == 0 {
+			c.warnUnnamed(p)
+			break
+		}
+		for _, id := range p.Names {
+			if id.Name != "_" {
+				objToIdent[id.Obj] = id
+			}
+		}
+	}
+
+	// remove used params
+	for id := range c.ctx.TypesInfo.Uses {
```

Comment:
We still do full `Uses` range here.
It can be more efficient to walk function body and mark params as used along the way,
looking up only usages present inside function itself.
Right now it ranges over whole package Uses set for every function.

If you don't feel like solving this issue right now, we can add this checker as experimental
along with TODO that explains that there is potential performance issue there.

Reply:
Well, below 1M(even 10M) iterations we're safe. I don't think this will explode in near future.
Sure, we can make an issue for that, but...I doubt that someone will mention a problem regarding this loop.
😉